.github/workflow-scripts/checkForReproducer.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+const NEEDS_REPRO_LABEL = 'Needs: Repro';
+const NEEDS_REPRO_MESSAGE = '| Missing Reproducible Example |';
+
+module.exports = async (github, context) => {
+  const issueData = {
+    issue_number: context.payload.issue.number,
+    owner: context.repo.owner,
+    repo: context.repo.repo,
+  };
+
+  const issue = await github.rest.issues.get(issueData);
+  const comments = await github.rest.issues.listComments(issueData);
+
+  const botComment = comments.data.find(comment =>
+    comment.body.includes(NEEDS_REPRO_MESSAGE),
+  );
+
+  let commentBodies = comments.data.map(comment => comment.body);
+  if (botComment) {
+    commentBodies = commentBodies.filter(body => body !== botComment.body);
+  }
+
+  const issueAndComments = [issue.data.body, ...commentBodies];
+  const issueAndCommentsUniq = [...new Set(issueAndComments)];
+
+  const user = issue.data.user.login;
+
+  const hasValidReproducer = issueAndCommentsUniq.some(body => {
+    const hasExpoSnackLink = containsPattern(
+      body,
+      `https?:\\/\\/snack\\.expo\\.dev\\/[^\\s)\\]]+`,
+    );
+    const hasGithubRepoLink = containsPattern(
+      body,
+      `https?:\\/\\/github\\.com\\/(${user})\\/[^/]+\\/?\\s?`,
```

Comment:
nit: This might be a bit too constrained. I.e what if a user posts a repro for a project that belongs to an org, or an existing repro from another user?

Reply:
It's a valid concern but my experience not that common. Implementing the bot so it would work when _any_ repo is present in the issue body would trigger too many false-positives.

I've seen more times that users tend to be "smart" about the repro and send the link to our repository like so:

![image](https://github.com/facebook/react-native/assets/39658211/a6de11f8-4524-499f-8063-75c86612720f)

Solid logic I guess - obviously the bug with Reanimated is in the Reanimated repo 🙄

Or even worse, have random repositories in the issue description:

![image](https://github.com/facebook/react-native/assets/39658211/e5574e30-4add-43e2-97f1-c270456d48d6)



Sure, I can implement as you suggest, but I'm pretty sure you won't be very happy about it long term